logstash-core/lib/logstash/agent.rb

```diff
@@ -106,7 +106,7 @@ def stop_background_services
   def configure_metric
     if collect_metric?
       @logger.debug("Agent: Configuring metric collection")
-      @metric = LogStash::Instrument::Metric.create(:root)
```

Comment:
If I understood your comment properly, you say that https://github.com/elastic/logstash/pull/4602/files#diff-bc1e9ea81caa54ea0d56abf4415fdf00R220 is basically creating the namespace?